scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -188,14 +188,18 @@
         (let [pod-status (.getStatus pod)
               ^V1ContainerStatus job-container-status (get-job-container-status pod-status)
               ; We leak mesos terminology here ('task') because of backward compatibility.
-              task-state (if (= (:state synthesized-state) :pod/succeeded)
+              succeeded? (= (:state synthesized-state) :pod/succeeded)
+              task-state (if succeeded?
                            :task-finished
                            :task-failed)
+              reason (if succeeded?
+                       :reason-normal-exit
+                       (or reason
+                           (container-status->failure-reason compute-cluster pod-name
+                                                             pod-status job-container-status)))
```

Comment:
Should we avoid overriding reason even when it is present?
```
(or reason
    (when succeeded?
      :reason-normal-exit)
    (container-status->failure-reason compute-cluster pod-name pod-status job-container-status))
```

Reply:
Good call, done